lib/stripe/webhook_plug.ex

```diff
@@ -159,30 +159,38 @@ if Code.ensure_loaded?(Plug) do
     end
 
     defp handle_event!(handler, %Stripe.Event{} = event) do
-      case handler.handle_event(event) do
-        {:ok, _} ->
-          :ok
-
-        :ok ->
-          :ok
-
-        {:error, reason} when is_binary(reason) ->
-          {:handle_error, reason}
-
-        {:error, reason} when is_atom(reason) ->
-          {:handle_error, Atom.to_string(reason)}
-
-        :error ->
-          {:handle_error, ""}
-
-        resp ->
-          raise """
-          #{inspect(handler)}.handle_event/1 returned an invalid response. Expected {:ok, term}, :ok, {:error, reason} or :error
-          Got: #{inspect(resp)}
-
-          Event data: #{inspect(event)}
-          """
-      end
+      :telemetry.span(~w[stripe webhook]a, %{}, fn ->
```

Comment:
Couldn't this be done at the WebServer Telemetry level? That was the feedback given from the OTEL folks

Reply:
I think yes, Phoenix and Plug already add their own events around requests, but these are very generic and we can do it better here by annotating with Stripe event type. Having this together with other part gives us more complete view of Stripe subsystem, or that's how I see it from monitoring perspective.

Reply:
I am with you! I see the point of adding `event.type` since it is domain-specific.

But wouldn't the WebServer handle the `handler_status` based on the response?

Reply:
btw I'm actually working on integrating OTEL and added Telemetry here to create spans and traces similar to how it's done in phoenix in [otel-erlang-contrib](https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex#L317-L344). I hope to have spans that include stripe api calls by name, I think that will be super helpful when browsing traces, but would be difficult to do using generic telemetry events from phoenix or other lib.

Reply:
I am not opposed to anything, but since these are architectural components, we are better off doing research and making sure we are following "standards" (OTEL practices for now) so we all benefit

I came across particular telemetry requirements that required the Operator to change their mindset and workflow to accomplish what they wanted; I am not saying that is the case here! I am just asking questions to make sure we are doing the right thing. I hope you understand.

This is a tricky domain; everyone wants their way of observing things...

Reply:
> But wouldn't the WebServer handle the handler_status based on the response?

Well, sort of. WebServer will report it in its own telemetry as completely different event so in a sense yes, but it's not always available. I think OTEL traces will be able to connect to parent span and present this info but other systems, for example Prometheus, won't.

Adding handler_status to this event makes it possible to monitor error rates by Stripe event type and use RED method to get useful insights with little effort, that's the main reason why I decided to add it here.

Alternatively we could add exactly same tracing to handler module that `WebhookHandler` is calling. It will work the same but this is additional effort to build custom observability versus having it provided ootb by stripity-stripe.

I personally prefer ootb experience because otherwise it's difficult to get in and do it right on your own. 

> I am not opposed to anything, but since these are architectural components, we are better off doing research and making sure we are following "standards" (OTEL practices for now) so we all benefit
> [...]

Yup, understood, happy to have this conversation and to look for best solution together, thanks for taking the time to discuss it too!